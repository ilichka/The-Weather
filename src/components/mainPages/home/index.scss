.home-page {
  display: flex;
  flex-direction: column;
  padding: 16px 16px 0;
  min-height: calc(100vh - 87px);
  justify-content: space-around;
  color: #07b299;

  .header {
    display: flex;
    flex-direction: row;
    justify-content: space-between;

    .header-info {
      display: flex;
      flex-direction: column;
      align-items: center;

      .city {
        font-size: 26px;
      }

      .temp {
        font-size: 46px;
        margin: 15px 0;
      }

      .status {
        padding: 7px 10px;
        border-radius: 20px;
        color: #fff;
        background: #95aea9;
      }
    }

    .header-icon {
      width: 120px;
      height: 120px;
    }
  }

  .info-row {
    display: flex;
    flex-direction: row;
    justify-content: space-between;

    .humidity-info,
    .wind-info,
    .pressure-info {
      align-items: center;
      display: flex;

      .humidity-icon,
      .wind-icon,
      .pressure-icon {
        width: 24px;
        height: 24px;
        margin-right: 5px;
      }
    }
  }

  .hourly-weather {
    display: flex;
    flex-direction: row;
    justify-content: space-between;

    .hour-block {
      display: flex;
      flex-direction: column;
      align-items: center;

      .hour-icon {
        width: 32px;
        height: 32px;
        margin: 12px 0;
      }

      .temp {
        color: #1e90ff;
        font-size: 32px;
      }
    }
  }

  .next-days {
    display: flex;
    flex-direction: column;

    .day-row {
      display: flex;
      flex-direction: row;
      align-items: center;
      margin-bottom: 15px;

      .day {
        min-width: 30%;
        font-size: 26px;
      }

      .day-icon {
        width: 50px;
        height: 50px;
        margin: auto;
      }

      .max-temp {
        margin-right: 15px;
        color: #1e90ff;
        font-size: 22px;
      }

      .min-temp {
        color: #95aea9;
      }
    }
  }
}

.dark-theme {
  background-color: black;

  .home-page {
    color: #8e05c2;

    .header-info {
      .status {
        color: #1597bb;
        background: #3e065f;
      }
    }

    .hourly-weather {
      .hour-block {
        .temp {
          color: #3d2c8d;
        }
      }
    }

    .next-days {
      .max-temp {
        color: #700b97;
      }

      .min-temp {
        color: #3d2c8d;
      }
    }
  }
}
