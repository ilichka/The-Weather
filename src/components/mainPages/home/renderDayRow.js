import { days } from "../../../utils/consts";

const getElemHTML = (weather, temp, day) => {
  return `
             <span class="day">${days[day]}</span>
            <img class="day-icon" src="http://openweathermap.org/img/wn/${
              weather[0].icon
            }.png" alt="day icon"/>
            <span class="max-temp">${Math.round(temp.max)}°</span>
            <span class="min-temp">${Math.round(temp.min)}°</span>
          `;
};

const createElem = (weather, temp, day) => {
  const elem = document.createElement("div");
  elem.classList.add("day-row");
  elem.innerHTML = getElemHTML(weather, temp, day);
  document.querySelector(".next-days").appendChild(elem);
};

const renderDayRow = (obj) => {
  for (let i = 0; i <= 2; i++) {
    const { weather, temp } = obj[i];
    const day = new Date().getDay() + 1 + i;
    createElem(weather, temp, day);
  }
};

export default renderDayRow;
