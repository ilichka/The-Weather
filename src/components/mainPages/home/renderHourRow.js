const getElemHTML = (temp, weather, time) => `<div class="hour-block">
            <span class="time">${time}:00 h</span>
            <img class="hour-icon" src="http://openweathermap.org/img/wn/${
              weather[0].icon
            }.png" alt="hour icon"/>
            <span class="temp">${Math.round(temp)}Â°</span>
            </div>
          `;

const renderHourRow = (obj) => {
  let row = ``;
  for (let i = 0; i <= 5; i++) {
    const { temp, weather } = obj[4 * i];
    const time = (new Date().getHours() + 4 * i) % 24;
    row += getElemHTML(temp, weather, time);
  }
  return row;
};

export default renderHourRow;
