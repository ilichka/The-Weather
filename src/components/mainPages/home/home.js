import "./index.scss";
import * as waterDropIcon from "../../../assets/img/waterdrop.svg";
import * as windIcon from "../../../assets/img/wind.svg";
import * as pressureIcon from "../../../assets/img/pressure.svg";
import { getLocation } from "../../../requester";
import renderDayRow from "./renderDayRow";
import renderHourRow from "./renderHourRow";
import { addNotification } from "../../../utils/notificationService/notificationService";

const getHomeHTML = (obj) => {
  const { city, temp, status, humidity, wind, icon, hourly, daily } = obj || {};
  return `
        <div class="header">
          <div class="header-info">
            <span class="city">${city}</span>
            <span class="temp">${temp}Â°</span>
            <span class="status">${status}</span>
          </div>
          <img class="header-icon" src="http://openweathermap.org/img/wn/${icon}.png"  alt="icon"/>
        </div>
        <div class="info-row">
          <div class="humidity-info">
            <img class="humidity-icon" src="${
              waterDropIcon.default
            }" alt="water drop"/>
            <span class="humidity">${humidity}%</span>
          </div>
          <div class="wind-info">
            <img class="wind-icon" src="${windIcon.default}" alt="wind"/>
            <span class="wind-speed">${wind}km/h</span>
          </div>
          <div class="pressure-info">
            <img class="pressure-icon" src="${
              pressureIcon.default
            }" alt="pressure"/>
            <span class="pressure">0.43mBar</span>
          </div>
        </div>
        <div class="hourly-weather">${renderHourRow(hourly)}</div>
        <div class="next-days">${renderDayRow(daily)}</div>
     `;
};

const initPage = async (element) => {
  const locations = await getLocation();
  if (!locations.cod) {
    const fragment = document.createElement("div");
    fragment.classList.add("home-page");
    fragment.innerHTML = getHomeHTML({
      city: locations.timezone,
      temp: Math.round(locations.current.temp),
      status: locations.current.weather[0].description,
      humidity: locations.current.humidity,
      wind: locations.current.wind_speed,
      icon: locations.current.weather[0].icon,
      hourly: locations.hourly,
      daily: locations.daily,
    });
    element.appendChild(fragment);
  } else {
    addNotification("error", locations.message);
  }
};

const homePage = () => {
  const page = document.createElement("div");

  initPage(page);

  return page;
};

export default homePage;
