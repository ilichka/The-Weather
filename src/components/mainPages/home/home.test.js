import { getHomeHTML } from "./home";

const generateTestData = () => {
  const data = {
    city: "city",
    country: "country",
    status: "status",
    temp: "temp",
    humidity: "humidity",
    wind: "wind",
    icon: "04d",
    hourly: [],
    daily: [],
  };
  for (let i = 0; i < 48; i++) {
    if (i % 6 === 0) {
      data.daily.push({
        weather: [{ icon: "04d" }],
        temp: { max: 8.67, min: 2.37 },
      });
    }
    data.hourly.push({ weather: [{ icon: "04d" }], temp: 7.99 });
  }
  return data;
};

describe("Home page snapshot", () => {
  it("Renders correctly template with mock data object", () => {
    const template = getHomeHTML(generateTestData());

    expect(template).toMatchSnapshot();
  });
});
