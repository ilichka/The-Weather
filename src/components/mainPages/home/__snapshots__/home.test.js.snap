// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page snapshot Renders correctly template with mock data object 1`] = `
"
        <div class=\\"header\\">
          <div class=\\"header-info\\">
            <span class=\\"city\\">city</span>
            <span class=\\"temp\\">temp°</span>
            <span class=\\"status\\">status</span>
          </div>
          <img class=\\"header-icon\\" src=\\"http://openweathermap.org/img/wn/04d.png\\"  alt=\\"icon\\"/>
        </div>
        <div class=\\"info-row\\">
          <div class=\\"humidity-info\\">
            <img class=\\"humidity-icon\\" src=\\"test-file-stub\\" alt=\\"water drop\\"/>
            <span class=\\"humidity\\">humidity%</span>
          </div>
          <div class=\\"wind-info\\">
            <img class=\\"wind-icon\\" src=\\"test-file-stub\\" alt=\\"wind\\"/>
            <span class=\\"wind-speed\\">windkm/h</span>
          </div>
          <div class=\\"pressure-info\\">
            <img class=\\"pressure-icon\\" src=\\"test-file-stub\\" alt=\\"pressure\\"/>
            <span class=\\"pressure\\">0.43mBar</span>
          </div>
        </div>
        <div class=\\"hourly-weather\\"><div class=\\"hour-block\\">
            <span class=\\"time\\">14:00 h</span>
            <img class=\\"hour-icon\\" src=\\"http://openweathermap.org/img/wn/04d.png\\" alt=\\"hour icon\\"/>
            <span class=\\"temp\\">8°</span>
            </div>
          <div class=\\"hour-block\\">
            <span class=\\"time\\">18:00 h</span>
            <img class=\\"hour-icon\\" src=\\"http://openweathermap.org/img/wn/04d.png\\" alt=\\"hour icon\\"/>
            <span class=\\"temp\\">8°</span>
            </div>
          <div class=\\"hour-block\\">
            <span class=\\"time\\">22:00 h</span>
            <img class=\\"hour-icon\\" src=\\"http://openweathermap.org/img/wn/04d.png\\" alt=\\"hour icon\\"/>
            <span class=\\"temp\\">8°</span>
            </div>
          <div class=\\"hour-block\\">
            <span class=\\"time\\">2:00 h</span>
            <img class=\\"hour-icon\\" src=\\"http://openweathermap.org/img/wn/04d.png\\" alt=\\"hour icon\\"/>
            <span class=\\"temp\\">8°</span>
            </div>
          <div class=\\"hour-block\\">
            <span class=\\"time\\">6:00 h</span>
            <img class=\\"hour-icon\\" src=\\"http://openweathermap.org/img/wn/04d.png\\" alt=\\"hour icon\\"/>
            <span class=\\"temp\\">8°</span>
            </div>
          <div class=\\"hour-block\\">
            <span class=\\"time\\">10:00 h</span>
            <img class=\\"hour-icon\\" src=\\"http://openweathermap.org/img/wn/04d.png\\" alt=\\"hour icon\\"/>
            <span class=\\"temp\\">8°</span>
            </div>
          </div>
        <div class=\\"next-days\\"> <div class=\\"day-row\\">
             <span class=\\"day\\">Thursday</span>
            <img class=\\"day-icon\\" src=\\"http://openweathermap.org/img/wn/04d.png\\" alt=\\"day icon\\"/>
            <span class=\\"max-temp\\">9°</span>
            <span class=\\"min-temp\\">2°</span>
            </div>
           <div class=\\"day-row\\">
             <span class=\\"day\\">Friday</span>
            <img class=\\"day-icon\\" src=\\"http://openweathermap.org/img/wn/04d.png\\" alt=\\"day icon\\"/>
            <span class=\\"max-temp\\">9°</span>
            <span class=\\"min-temp\\">2°</span>
            </div>
           <div class=\\"day-row\\">
             <span class=\\"day\\">Saturday</span>
            <img class=\\"day-icon\\" src=\\"http://openweathermap.org/img/wn/04d.png\\" alt=\\"day icon\\"/>
            <span class=\\"max-temp\\">9°</span>
            <span class=\\"min-temp\\">2°</span>
            </div>
          </div>
     "
`;
